app/Console/Commands/RebuildQueryserviceData.php

```diff
@@ -0,0 +1,153 @@
+<?php
+
+namespace App\Console\Commands;
+
+use Illuminate\Console\Command;
+use Illuminate\Support\Facades\Http;
+use Illuminate\Support\Facades\Log;
+use App\Wiki;
+use App\WikiSetting;
+use App\QueryserviceNamespace;
+use App\Jobs\SpawnQueryserviceUpdaterJob;
+
+class RebuildQueryserviceData extends Command
+{
+    private const NAMESPACE_ITEM = 120;
+    private const NAMESPACE_PROPERTY = 122;
+    private const NAMESPACE_LEXEME = 146;
+
+    protected $signature = 'wbs-qs:rebuild {--domain=*} {--chunkSize=50} {--sparqlUrlFormat=http://queryservice.default.svc.cluster.local:9999/bigdata/namespace/%s/sparql}';
+
+    protected $description = 'Rebuild the queryservice data for a certain wiki or all wikis';
+
+    protected int $chunkSize;
+    protected string $apiUrl;
+    protected string $sparqlUrlFormat;
+
+    public function handle()
+    {
+        $this->chunkSize = intval($this->option('chunkSize'));
+        $this->sparqlUrlFormat = $this->option('sparqlUrlFormat');
+        $this->apiUrl = getenv('PLATFORM_MW_BACKEND_HOST').'/w/api.php';
+
+        $wikiDomains = $this->option('domain');
+        $exitCode = 0;
+
+        $wikis = count($wikiDomains) !== 0
+            ? Wiki::whereIn('domain', $wikiDomains)->get()
+            : Wiki::query()->get();
+
+        foreach ($wikis as $wiki) {
+            try {
+                $entities = $this->getEntitiesForWiki($wiki);
+                $sparqlUrl = $this->getSparqlUrl($wiki);
+            } catch (\Exception $ex) {
+                Log::error(
+                    'Failed to get prerequisites for enqueuing wiki '.$wiki->domain.', will not dispatch jobs.'
+                );
+                Log::error($ex);
+                $exitCode = 1;
+                break;
+            }
+
+            $entityChunks = array_chunk($entities, $this->chunkSize);
+            foreach ($entityChunks as $entityChunk) {
+                dispatch(
```

Comment:
This is probably hard to circumvent because of the nature of this batching technique but I think it can get weird if you have a large wiki and the batch job fails for some reason for some (or all) of the batches - suddenly you have to deal with a lot of batches.

For example right now locally for me somethingTM is wrong, 4 jobs get dispatched, all of them fail and it gets hard to look through the logs. Still I think this might be the desired behaviour, I just wanted to flag this

Reply:
I was wondering the same, but couldn't come up with something really good.

Two things I thought of:
- introduce a failure allowance and instead of continuing, bail when too many wikis have failed (this would happen when fetching the data above though)
- Introduce a `limit` and `offset` parameter and allow running the command in batches instead of for ALL wikis at once

Reply:
yeah seems tricky... fwiw I'm totally okay with how it is right now, as long as we're aware how it works we should be fine